tests/suites/test_suite_psa_crypto.function

```diff
@@ -301,7 +301,7 @@ uint64_t parse_binary_string(data_t *bin_string)
     uint64_t result = 0;
     TEST_LE_U(bin_string->len, 8);
     for (size_t i = 0; i < bin_string->len; i++) {
-        result |= ((bin_string->x)[i]) << (i * 8);
+        result = (uint64_t) result | bin_string->x[i];
```

Comment:
Ugh, no, I still can't get this right: it should be
```
        result = result << 8 | bin_string->x[i];
```

Someone with a working brain please check what I write!

Reply:
(Please check with a debugger that the library function receives the expected value.)

Reply:
And since we went through so many iterations, I'm requesting a unit test that checks that we got it right this time. I really don't want to end up in a situation later where we try to debug why our new code using an integer input isn't working, and it turns out that we weren't passing the data we were expecting to the library function because of a bug in a test function.

Something like
```
parse_binary_string:"012345":0x12345
```
Note that there's a limitation in the test framework that integer constants are of type `int`, so test cases have to be less than 2^31 for now.

Reply:
Actually, let's not reinvent the wheel! Let's use `strtoull` (with base=16).

We currently don't use `strtoull`, but we use `long long` and `strtoul`, so I think we can safely assume that `strtoull` exists.